### DISCUSSION TITLE
Not updating changes quickly - How can we improve the refresh time?

### DISCUSSION THREAD

**Discussion by Jurgem** [samemory/homebridge-eufy-security-control#43] [General]
The status of entry sensor is not updating, when I open (or close) a door it stay in the same status (opened or closed). It only change when I reboot home bridge

Awesome plugin!

**Comment by samemory** [comment]
Right now I believe it's only updating status when it refreshes the data according to the polling interval. The default is 10 mins but in the config you can change it. Haven't tested a different polling interval though personally. 
Hope to find a better solution for this, it's kind of annoying.

**Comment by lenoxys** [comment]
In fact, we do report the open/close but relying on the polling interval which is 10min by default or push notification. For this last if you select a security profile (home, away, cust), you have to enable "mouvement reaction" and push notification. If you do so, the status is updated right away (but you will receive notification on your phone as well each time the status change if you have the application installed ).

**Comment by Jurgem** [comment]
i see, when I activate the push notification in Eufy security, it update instantly. Ok, so when im in home I don't want to eufy send me notifications, so I think that the minimum is to put the polling interval  = 1?

**Comment by schliemann** [comment]
I actually don't think the pollingIntervalMinutes trigger will do anything regarding update of device state like open/close for an entry sensor? RefreshData updates the stations and devices inside the clientlibrary. But I don't think we look at them again after discoverDevices has run

RefreshData: https://github.com/samemory/homebridge-eufy-security/blob/0729592aec37e8a73184253b98636884926c9eb7/src/platform.ts#L308-L327

discoverDevices: https://github.com/samemory/homebridge-eufy-security/blob/0729592aec37e8a73184253b98636884926c9eb7/src/platform.ts#L123

**Comment by lenoxys** [comment]
Hello,

You right ! It's done because HomeKit will triggered the event of getting or if we receive a push notification.

I think we shouldn't rely on polling the value, I can see I can get ban for a day because I generated too much requests.

Regards,
Aurelien

**Comment by schliemann** [comment]
How many requests did you send? 🤔😅

**Comment by lenoxys** [comment]
A lot, but since the push notification it's more easy to debug without doing too much request. I've to setup a dedicated account link to the first one to avoid blocking everything.

**Comment by schliemann** [comment]
https://user-images.githubusercontent.com/10818044/121748353-17d65a00-cb09-11eb-878c-aa2a86371c44.MOV


I just tested this and can verify that the push notifications for entry sensor works.

**Comment by schliemann** [comment]
I think refreshData could be used to discover newly added devices to the hub. But it is a "nice to have" feature. And probably not worth implementing :)

**Comment by lenoxys** [comment]
Yeah but the Eufy app is really noisy when enabling push while at home. Does p2p would work without the need of enabling push ?

**Comment by samemory** [comment]
It does look like there's another way to get updates using p2p based on eufy-security-client code. Looks like there are some emitters that could be used but just haven't had time to dig into it yet. I could be completely wrong. Maybe the examples in this library would be helpful: https://github.com/bropat/eufy-security-ws

**Comment by OrigSorceror** [comment]
Not able to get the Eufy T8910 motion sensors working with the plugin at all.  It detects the sensors and grabs the details from the sensors and adds them in to HomeBridge OK, but it does not change state when motion is detected. Even with "React to motion" and "Push Notifications" enabled for the sensor in the Eufy App. Homebridge state remains as no motion detected (but i get the Eufy notifications constantly in the Eufy App) 

The main reason i want these motion sensors working is to automate turning on lights in the home/area where the motion sensor is located. (i currently have this achieved in two important locations using EVE motion sensors), but i have 4 Eufy sensors for my Eufy Alarm system.

**Reply by lenoxys** [reply]
this should be fixed now

**Reply by OrigSorceror** [reply]
yes i can confirm it now works.. Now if only there was a way to do away witht eh constant notifications in the eufy app :/

**Reply by lenoxys** [reply]
I took the bet to disable notification from eufy app and rely on the home one.

**Reply by OrigSorceror** [reply]
yes need the push notification set for the sensor in the Eufy App but disable all notifications and sounds from the eufy app within the settings of iPhone. Also disabled the notifications in the Home app once confirmed it was still working and it works flawlessly.. no if only i can work out what is causing the delay in response (sometimes upto 2 minutes) for the automation's to trigger.  But i am thinking this is likely to do with the raspberry Pi 3b i am running homebridge on as the EVE motions sensors also have the delay. My lights, fans and shutters are controlled via a Clipsal CBus SHAC (wiser 3) and linked to HomeKit via Homebridge and cgate.

**Reply by lenoxys** [reply]
I run on a VM I got notifications right away ( which is really useful while troubleshooting)

**Reply by OrigSorceror** [reply]
Oh i get the notifications right away. No issues there.  It is just a delay between notification and triggering of the associated automation, i.e once motion detected turn on lights.  Sometimes it works instantly other times it can take up to two minutes for the Automation to run, I notice a lot of activity from the polling of various events in Hombridge log. Also the RPi3b has only 1 gig memory and a 16gig microSD, So it is likely that a combination of the Micro SD write speed may be too slow and the memory of the Pi 3b is also being slow. I may grab a Pi4b 4gig and a 32 Gig SD and try that.. Granted the CBus network only runs at 10mbs but it still should not take that long.. THe CBus network is running on an entirely separate Network than my Home data network (for Security) with the Pi3b being the only permanent bridge between the 2 physical networks (i also have a 3rd router that i configured to bridge the network in case i wish to do some programming of the Wiser3 or the CBus devices).  Also sometimes Siri times out with getting no response from the device. It does eventually turn on though. So i think its a latency issue with HomeBridge.  Worse case scenario i can always run an old laptop as the HomeBridge machine ( i got a shed full of old Dells) or a full on Dell Poweredge blade server.. But that s overkill LOL
